stripe/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -166,11 +167,11 @@ internal class DefaultFlowController internal constructor(
 
         paymentOptionLauncher(
             PaymentOptionContract.Args(
-                paymentIntent = initData.paymentIntent,
+                stripeIntent = initData.stripeIntent,
                 paymentMethods = initData.paymentMethods,
                 sessionId = sessionId,
                 config = initData.config,
-                isGooglePayReady = initData.isGooglePayReady,
+                isGooglePayReady = initData.isGooglePayReady && initData.stripeIntent is PaymentIntent,
```

Comment:
This should not return ready unless the stripeIntent is a paymentIntent

Reply:
Right, isn't that what this is doing?